docs/reference/settings/security-hash-settings.asciidoc

```diff
@@ -12,32 +12,46 @@ hashing algorithm by setting the <<static-cluster-setting,static>>
+| `bcrypt7`               | | | Uses `bcrypt` algorithm with salt generated in 128 rounds.
+| `bcrypt8`               | | | Uses `bcrypt` algorithm with salt generated in 256 rounds.
+| `bcrypt9`               | | | Uses `bcrypt` algorithm with salt generated in 512 rounds.
+| `pbkdf2`                | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
                              pseudorandom function using 10000 iterations.
-| `pbkdf2_1000`       | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
-                             pseudorandom function using 1000 iterations.
-| `pbkdf2_10000`      | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
-                             pseudorandom function using 10000 iterations.
-| `pbkdf2_50000`      | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
-                             pseudorandom function using 50000 iterations.
-| `pbkdf2_100000`     | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
-                             pseudorandom function using 100000 iterations.
-| `pbkdf2_500000`     | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
-                              pseudorandom function using 500000 iterations.
-| `pbkdf2_1000000`    | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
-                             pseudorandom function using 1000000 iterations.
-| `noop`,`clear_text` | | | Doesn't hash the credentials and keeps it in clear text in
+| `pbkdf2_1000`           | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
+                             pseudorandom function over SHA512 hash using 1000 iterations.
+| `pbkdf2_10000`          | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
+                             pseudorandom function over SHA512 hash using 10000 iterations.
+| `pbkdf2_50000`          | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
+                             pseudorandom function over SHA512 hash using 50000 iterations.
+| `pbkdf2_100000`         | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
+                             pseudorandom function over SHA512 hash using 100000 iterations.
+| `pbkdf2_500000`         | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
+                              pseudorandom function over SHA512 hash using 500000 iterations.
+| `pbkdf2_1000000`        | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
+                             pseudorandom function over SHA512 hash using 1000000 iterations.
+| `pbkdf2_stretch`        | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
+                             pseudorandom function over SHA512 hash using 10000 iterations.
+| `pbkdf2_stretch_1000`   | | | Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a
```

Comment:
I think we need to add some hint to the stretching here, otherwise folks would be confused as to why we added  all these new hashers that do the same with the existing ones.

Reply:
@bytebilly do you have any suggestions for a short and meaningful description?

Reply:
What if we just describe _what_ it does, and defer the _why_ to a note? I see other descriptions don't give the _why_ too.

> Uses `PBKDF2` key derivation function with `HMAC-SHA512` as a pseudorandom function over `SHA512` hash using 10000 iterations, hashing the initial input with `SHA512` first.

The note can be reported at the end of the table, stating that this is the value that should be used in FIPS 140-2 environments to guarantee compatibility.

What do you think?

Reply:
> What if we just describe what it does, and defer the why to a note?

Apologies if this was unclear, I didn't mean to imply that we need to explain why ( at least not in this doc ) with my 

> otherwise folks would be confused as to why we added all these new hashers that do the same with the existing ones.

I don't think we should add the `why` here. We can add a reference to this hasher in our FIPS 140 docs. 

+1 for the text maybe add an `after` : `...., after hashing the initial input with SHA512 first` ?

Reply:
I find the order of this sentence to be strange - it explains the hashing process in reverse.
My suggestion:
```
| `pbkdf2_stretch_{n}`    | | | First stretches the input password by passing it through a SHA512 hash
                             and then applies the `PBKDF2` key derivation function with `HMAC-SHA512` as a
                             pseudorandom function using {n} iterations.
```